doc/source/whatsnew/v0.23.0.txt

```diff
@@ -415,6 +415,7 @@ I/O
 - Bug in :func:`read_sas` where a file with 0 variables gave an ``AttributeError`` incorrectly. Now it gives an ``EmptyDataError`` (:issue:`18184`)
 - Bug in :func:`DataFrame.to_latex()` where pairs of braces meant to serve as invisible placeholders were escaped (:issue:`18667`)
 - Bug in :func:`read_json` where large numeric values were causing an ``OverflowError`` (:issue:`18842`)
+- Bug in :func:`DataFrame.to_parquet` exception is thrown if write destination is S3 (:issue:`19134`)
```

Comment:
where exception was raised

Reply:
There are several possible exceptions in the fail chain.

3 different components

 - S3Filesystem, 
 - pyarrow writer, 
 - fastparquet reader & writer.

### pyarrow - write attempt
FileNotFoundException or ValueError (depends on if file exists in S3 or not).

### fastparquet - read attempt
Exception in attempting to concat str and S3File

### fastparquet - write attempt
Exception in attempting to open path using default_open

Reply:
where an exception was raised if the write destination is S3.